### PR TITLE
samples: matter: add support for nRF54L10 and nRF54L05

### PR DESCRIPTION
**Pull request by maciejbaczmanski**
Add support for nRF54L10 and nRF54L05 in OpenThread

Add support for nRF54L10 in Matter

### CODE REVIEW

**Comment by maciejbaczmanski** [comment]
CI will fail until RD libs are added for 54l10 and 54l05 targets

File: samples/openthread/cli/README.rst
Comment (annwoj) [inline samples/openthread/cli/README.rst:102]:
```suggestion
     The ``usb`` snippet does not support the ``nrf54l15dk/nrf54l15/cpuapp`` and ``nrf54l15dk/nrf54l10/cpuapp`` board targets.
```

File: samples/openthread/cli/README.rst
Comment (annwoj) [inline samples/openthread/cli/README.rst:129]:
```suggestion
     The ``low_power`` snippet does not support the ``nrf54l15dk/nrf54l10/cpuapp`` board target.
```

File: samples/openthread/coprocessor/README.rst
Comment (divipillai) [inline samples/openthread/coprocessor/README.rst:114]:
```suggestion
     The ``usb`` snippet is not supported for the ``nrf54l15dk/nrf54l15/cpuapp`` and the ``nrf54l15dk/nrf54l05/cpuapp`` board targets.
```
To fix the doc build error.

File: samples/openthread/cli/boards/nrf54l15dk_nrf54l10_cpuapp.overlay
Comment (LuDuda) [inline samples/openthread/cli/boards/nrf54l15dk_nrf54l10_cpuapp.overlay:10]: Should it be 1022 or 1012? Or in other words, how you end up in number 1022?

File: samples/openthread/cli/boards/nrf54l15dk_nrf54l10_cpuapp.overlay
Comment (nordicjm) [inline samples/openthread/cli/boards/nrf54l15dk_nrf54l10_cpuapp.overlay:7]:
```suggestion
```

**Review by nordicjm — APPROVED** [review]

File: samples/openthread/coprocessor/README.rst
Comment (maciejbaczmanski) [reply samples/openthread/coprocessor/README.rst]: fixed similar as in cli sample to align with @annwoj 's comment

File: subsys/net/openthread/Kconfig.defconfig
Comment (kkasperczyk-no) [inline subsys/net/openthread/Kconfig.defconfig:172]: Maybe it would be better to replace all listed boards with `SOC_SERIES_NRF54LX`?

**Review by kkasperczyk-no — APPROVED** [review]

File: subsys/net/openthread/Kconfig.defconfig
Comment (maciejbaczmanski) [reply subsys/net/openthread/Kconfig.defconfig:172]: CSL accuracy and uncertainty is directly correlated with clock accuracy on the DK, so we cannot use SOC related configs. unfortunately I didn't find any Kconfig that is common for all nrf54l15DK based targets

**Review by ArekBalysNordic — APPROVED** [review]

**Review by annwoj — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

File: subsys/net/openthread/Kconfig.defconfig
Comment (maciejbaczmanski) [reply subsys/net/openthread/Kconfig.defconfig]: @adigie checked that we can use BOARD_NRF54L15DK . Thanks!

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:519]: Remove the changelog entries from this PR and add them as a suggestion in my https://github.com/nrfconnect/sdk-nrf/pull/19240.

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by kkasperczyk-no — COMMENTED** [review]
I think we should update snippets/matter-diagnostic-logs/boards and snippets/matter-diagnostic-logs/snippet.yml as well.

File: west.yml
Comment (ArekBalysNordic) [inline west.yml:152]: Just leaving a comment for you to not forget about it :)

File: samples/matter/light_bulb/sample.yaml
Comment (maciejbaczmanski) [reply samples/matter/light_bulb/sample.yaml:16]: I'd at least let it run for all samples overnight as we are waiting for RD libs anyways, if support is not necessary in all samples, I'll remove some tomorrow morning

File: doc/nrf/protocols/matter/end_product/security.rst
Comment (LuDuda) [reply doc/nrf/protocols/matter/end_product/security.rst:146]: If we talk about SoC a in this table, and not DK. I would leave nRF54L10. For sections where we mention DKs, nRf54L10 emulation on nRF54L15 DK makes sense.

File: doc/nrf/protocols/matter/getting_started/hw_requirements.rst
Comment (LuDuda) [reply doc/nrf/protocols/matter/getting_started/hw_requirements.rst:862]: @maciejbaczmanski would you be able to still add it?

File: samples/matter/light_bulb/sample.yaml
Comment (LuDuda) [reply samples/matter/light_bulb/sample.yaml:16]: No, the plan is just to enable it for Template and possibly for Smoke CO Alarm sample. @maciejbaczmanski 
Else RAM requirements might not be met.

File: doc/nrf/protocols/matter/getting_started/hw_requirements.rst
Comment (maciejbaczmanski) [reply doc/nrf/protocols/matter/getting_started/hw_requirements.rst:862]: @ArekBalysNordic just started working on that

File: samples/matter/light_bulb/sample.yaml
Comment (maciejbaczmanski) [reply samples/matter/light_bulb/sample.yaml]: yeah, my mistake. left changes only for template and smoke co alarm

File: samples/matter/light_bulb/sysbuild/mcuboot/boards/nrf54l15dk_nrf54l10_cpuapp.conf
Comment (maciejbaczmanski) [reply samples/matter/light_bulb/sysbuild/mcuboot/boards/nrf54l15dk_nrf54l10_cpuapp.conf]: removed as it was copied from l15

File: doc/nrf/protocols/matter/getting_started/hw_requirements.rst
Comment (ArekBalysNordic) [reply doc/nrf/protocols/matter/getting_started/hw_requirements.rst:862]: Done.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (maciejbaczmanski) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: removed and added suggestion

File: doc/nrf/protocols/matter/end_product/security.rst
Comment (maciejbaczmanski) [reply doc/nrf/protocols/matter/end_product/security.rst]: fixed

**Review by LuDuda — APPROVED** [review]

File: snippets/matter-diagnostic-logs/boards/nrf54l15dk_nrf54l10_cpuapp.overlay
Comment (kkasperczyk-no) [inline snippets/matter-diagnostic-logs/boards/nrf54l15dk_nrf54l10_cpuapp.overlay:9]: 1CC0=7360, after decreasing it should be 0xCC0

Comment (kkasperczyk-no) [reply snippets/matter-diagnostic-logs/boards/nrf54l15dk_nrf54l10_cpuapp.overlay:9]: Also, shouldn't it be 0x200**3**F340 instead of 0x2002F340?

File: snippets/matter-diagnostic-logs/boards/nrf54l15dk_nrf54l10_cpuapp.overlay
Comment (kkasperczyk-no) [inline snippets/matter-diagnostic-logs/boards/nrf54l15dk_nrf54l10_cpuapp.overlay:49]: I believe here also it should be 0x3F340

File: snippets/matter-diagnostic-logs/boards/nrf54l15dk_nrf54l10_cpuapp.overlay
Comment (kkasperczyk-no) [reply snippets/matter-diagnostic-logs/boards/nrf54l15dk_nrf54l10_cpuapp.overlay:9]: Nvm, it's ok, discussed f2f

Comment (maciejbaczmanski) [reply snippets/matter-diagnostic-logs/boards/nrf54l15dk_nrf54l10_cpuapp.overlay]: 0xCC0 was indeed incorrect. fixed

File: snippets/matter-diagnostic-logs/boards/nrf54l15dk_nrf54l10_cpuapp.overlay
Comment (maciejbaczmanski) [reply snippets/matter-diagnostic-logs/boards/nrf54l15dk_nrf54l10_cpuapp.overlay:49]: as discussed offline, 0x2F340 is correct

File: snippets/matter-diagnostic-logs/boards/nrf54l15dk_nrf54l10_cpuapp.overlay
Comment (kkasperczyk-no) [inline snippets/matter-diagnostic-logs/boards/nrf54l15dk_nrf54l10_cpuapp.overlay:36]:
```suggestion
			end_user_logs_retention: retention@8c0 {
```

File: samples/matter/smoke_co_alarm/sysbuild/mcuboot/boards/nrf54l15dk_nrf54l10_cpuapp.conf
Comment (adigie) [inline samples/matter/smoke_co_alarm/sysbuild/mcuboot/boards/nrf54l15dk_nrf54l10_cpuapp.conf:7]:
```suggestion
# TODO: Workaround Fprotect is not supported on nRF54l10 yet.
```

File: snippets/matter-diagnostic-logs/boards/nrf54l15dk_nrf54l10_cpuapp.overlay
Comment (maciejbaczmanski) [reply snippets/matter-diagnostic-logs/boards/nrf54l15dk_nrf54l10_cpuapp.overlay]: fixed

File: samples/matter/smoke_co_alarm/sysbuild/mcuboot/boards/nrf54l15dk_nrf54l10_cpuapp.conf
Comment (maciejbaczmanski) [reply samples/matter/smoke_co_alarm/sysbuild/mcuboot/boards/nrf54l15dk_nrf54l10_cpuapp.conf]: it should not even be here and in l15's overlay... god catch, thanks :)

**Comment by maciejbaczmanski** [comment]
@nrfconnect/ncs-co-networking @nrfconnect/ncs-code-owners could you review please? We are just waiting for deployment of RD libs (manifest update, will replace [[DNM] fake nrf54l10/05 802154 libs for twister builds](https://github.com/nrfconnect/sdk-nrf/pull/19209/commits/579bc88a90c38437aad22f3a691d0a9a306e9d0d) and should be ready to merge

File: doc/nrf/protocols/matter/getting_started/hw_requirements.rst
Comment (peknis) [inline doc/nrf/protocols/matter/getting_started/hw_requirements.rst:40]:
```suggestion
   The nRF54L15 supports DFU with image compression, which may eliminate the need for external flash.
```

File: doc/nrf/protocols/matter/getting_started/hw_requirements.rst
Comment (peknis) [inline doc/nrf/protocols/matter/getting_started/hw_requirements.rst:222]:
```suggestion
      You can use the table as a reference for the :ref:`nRF54L15 DK (emulating nRF54L10) <programming_board_names>` (:ref:`nrf54l15dk/nrf54l10/cpuapp <zephyr:nrf54l15dk_nrf54l15>`).
```

**Review by peknis — APPROVED** [review]
Approved with minor nits.

**Review by rlubos — APPROVED** [review]
Approving, provided the DNM commit is dropped.

File: doc/nrf/protocols/matter/getting_started/hw_requirements.rst
Comment (maciejbaczmanski) [reply doc/nrf/protocols/matter/getting_started/hw_requirements.rst]: fixed

File: west.yml
Comment (maciejbaczmanski) [reply west.yml]: changed to real PR

**Pull request merged**
